#! /usr/bin/env python

PACKAGE='raw_perception_tests'
import roslib; roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

#        Name                             Type         Description                                   Default Min Max
gen.add("normal_max_depth_change_factor", double_t, 0, "Normal estimation max depth change factor",  0.02,   0,  0.1  )
gen.add("normal_smoothing_size",          double_t, 0, "Normal estimation smoothing size",           10.0,   0,  30.0 )
gen.add("min_inliers",                    int_t,    0, "Minimum inliers",                            1000,   0,  10000)
gen.add("angular_threshold",              double_t, 0, "Angular threshold (degrees)",                3.0,    0,  10   )
gen.add("distance_threshold",             double_t, 0, "Distance threshold",                         0.02,   0,  0.2  )
gen.add("maximum_curvature",              double_t, 0, "Maximum curvature",                          0.001,  0,  0.1  )
gen.add("refinement_threshold",           double_t, 0, "Refinement threshold",                       0.02,   0,  0.2  )
gen.add("refinement_depth_independent",   bool_t,   0, "Refinement depth independent",               True)

#        Name                              Type         Description                        Default Min   Max 
gen.add("apply_plane_constraints",         bool_t,   0, "Apply plane constraints",         False)
gen.add("plane_normal_angular_threshold",  double_t, 0, "Plane normal angular threshold",  20,     0,    90    )
gen.add("plane_normal_x",                  double_t, 0, "Plane normal x",                  0.1,    -1.0, 1.0   )
gen.add("plane_normal_y",                  double_t, 0, "Plane normal y",                  0.1,    -1.0, 1.0   )
gen.add("plane_normal_z",                  double_t, 0, "Plane normal z",                  0.1,    -1.0, 1.0   )

exit(gen.generate(PACKAGE, "dominant_plane_extractor_test", "OrganizedDominantPlaneExtractorTest"))
